src/service/admin/media/media_commands.rs

```diff
@@ -139,14 +139,19 @@ pub(crate) async fn delete(
 
 pub(crate) async fn delete_list(body: Vec<&str>) -> Result<RoomMessageEventContent> {
 	if body.len() > 2 && body[0].trim().starts_with("```") && body.last().unwrap().trim() == "```" {
-		let mxc_list = body.clone().drain(1..body.len() - 1).collect::<Vec<_>>();
+		let mxc_list = body
+			.clone()
+			.drain(1..body.len().checked_sub(1).unwrap())
+			.collect::<Vec<_>>();
 
-		let mut mxc_deletion_count = 0;
+		let mut mxc_deletion_count: u32 = 0;
```

Comment:
Any reason not to make this a `usize`? Not that it's very likely anyone will be deleting 4 billion media files, but I don't see any benefit in restricting it like this.